(maint) add default and parallel builds.

nanliu opened the pull request:
This reduces the number of builds and speed up builds by increasing number of parallel plugin builds. We only need mac and linux builds when releasing, so it is no longer the default.

kindermoumoute commented on scripts/build_plugins.sh:
```diff
@@ -0,0 +1,61 @@
+
+OS=$(uname -s)
+if [[ "${OS}" == "Darwin" ]]; then
+  p=$(type -p sysctl > /dev/null && sysctl -n hw.ncpu || echo "1")
```
Is this possible to still have an option to manually define p? Some developer may prefer to wait longer but not have their laptop frozen.

nanliu commented:
Updated PR to show os/arch when building plugins:
```
2016-10-12 00:08:59 UTC [     info] building plugins for linux/amd64 in 2 parallels
2016-10-12 00:08:59 UTC [     info] building snap-plugin-collector-anothermock1 for linux/amd64
...
```

I added an option to allow you to specify number of parallel builds.
```
BUILD_JOBS=2 make plugins
```

kindermoumoute commented:
LGTM
